### PR TITLE
fix: nft table and avatar selection filtering by asset/collection

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

tl;dr us not being able to go join from portfolio yet because of the current discrepencies WRT NFT collection addresses in unchained and from Zapper/Alchemy (most likely because of proxies) shouldn't block us from using selectors.

Introducing `makeSelectNftItemsWithCollectionSelector`, which does precisely what it says: create a selector, which itself is created using a selector creator from RTK `getNftUserTokens` endpoints.

This allows the search by NFT name / Id / collection name / address to be great again.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Testing

- Search by NFT name, id, collection name and address are working in NFT table and avatar selection
## Screenshots (if applicable)

image
image
image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4623** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4623" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4623?utm_source=stack-comment).

**Review by 0xApotheosis — APPROVED** [review]
Filtering is indeed great again 🤝
